### DISCUSSION TITLE
Support

### DISCUSSION THREAD

**Discussion by kylecorry31** [kylecorry31/Trail-Sense#2098] [New Feature Ideas]
A place to ask questions about Trail Sense, understand how to use the tools, or raise bugs.

**Comment by SourceOfDeath** [comment]
Hi, I'm amazed by the functionality of your program. I couldn't use the map creation function from ready image. I select the desired image (jpg) and nothing happens, there are no error messages either. I checked that the absence of national characters in the full file name has no effect. I also checked the functionality with images of different sizes: 5000*3500 does not open. Trail-sense takes photos fine, but I don't have any paper maps. Perhaps there are unspecified restrictions on image properties that are not specified in the documentation?

**Reply by kylecorry31** [reply]
Hi, I'm sorry you are having issues with Trail Sense. Can you provide me the following information so I can try to figure out why that's not working for you:

1. Which version of Trail Sense are you using? (Tools > Settings > Version)
2. What is your phone's manufacturer and android version? (ex. Samsung, Android 13)
3. If you are comfortable doing so, could you share one of the maps with me? Either here or at trailsense@protonmail.com

 Unfortunately I haven't seen that issue before, and it is working fine for me, so it may take a little bit to figure out what is going on.

**Reply by SourceOfDeath** [reply]
You surprisingly fast! I always stuck into errors, that no one has seen before me.

1. My version is 6.9.0
2. My brick is Xiaomi Poco X6 5G 8Gb/256Gb, Android 13 TKQ1.221114.001
3. In case the origin of the files is important, I created them in the SAS Planet program. It automatically downloads pieces of maps from free services and stitches them together. 
![lg blocks](https://github.com/user-attachments/assets/cf584074-bf3a-40b0-8805-514816014f75)
I'll paste one file here, the rest are too big. I upload it:
[https://limewire.com/d/lIB7u#qoia6k4wz8](https://limewire.com/d/lIB7u#qoia6k4wz8)
They expires in 7 days.

Could You give me a picture that Trail Sense will definitely open (not necessarily a map, just for test)?

**Reply by kylecorry31** [reply]
Thank you for all those details. The first thing I notice are those files are really big (I don't have issues importing them though), which could cause an issue on some devices (though I would expect if the device runs out of memory you would see a crash - I've seen that before). Can you try it with the "Tools > Settings > Photo Maps > Reduce photo resolution"  setting both enabled and disabled? I'm guessing you want that setting off because these maps are so detailed, but the behavior may help me determine what's going on.

There shouldn't be anything that fixes it in 6.10.0, but that update is avalable everywhere you can find Trail Sense just to confirm it wasn't an issue with 6.9.0 that was unknowingly fixed.

Here's some sample maps:
- PDF: https://caltopo.com/p/014GS
- Random photo (draft survival guide art) that works for me:
![PXL_20250503_121716419](https://github.com/user-attachments/assets/80336672-fe89-41f1-acbe-5e26a9f218d4)

**Reply by SourceOfDeath** [reply]
I did not set up the program. I have now checked the settings - "Tools > Settings > Photo Map > Reduce photo resolution" is enabled. I sent both the pdf and the sketch to my phone. Both with resolution reduction turned off and resolution reduction turned on, the program does not open both files.

The application uses 18 MB, 2.90 GB of RAM and 4.0 GB of swap are available on the device.

We can conclude that the device is to blame. But other programs open these files.

**Reply by kylecorry31** [reply]
That is odd, I'll keep looking for a cause and may send a nightly build here if I find something.

**Reply by SourceOfDeath** [reply]
I remembered that I have LD Player. I have installed Trail Sense into it. When opening image, Trail Sense immediately asks for name of new map and suggests as default file name, then proceeds to calibrate new map.

**Reply by SourceOfDeath** [reply]
I opened Your pdf with LibreraFD application from F-Droid. In the phone settings, I looked at "Settings-applications-permissions-application permissions-files-applications with this permission": Librera, Explorer and Total Commander got this permission, but Trail Sense did not. Perhaps this is a "feature" of the Chinese.

**Reply by kylecorry31** [reply]
Thank you, that does not sound like a standard Android permission, so it must be something that Xiaomi is doing. I'll need to do more research on how that works and integrates with the built in android file picker that Trail Sense uses.

I created a new issue to track this for Xiaomi devices: https://github.com/kylecorry31/Trail-Sense/issues/2930

**Reply by kylecorry31** [reply]
Hm, I just tried a Xiaomi device on browserstack (android 13) and it worked fine, so I'm not convinced this is a Xiaomi permissions issue. I only have a free trial, so I quickly tested, saw it worked, and then lost access, so I wasn't able to test with your images (just the ones in the browserstack gallery). 

Can you provide the following for me:
1. Where did you download Trail Sense from?
2. Can you try to download the nightly build and see if that has the same problem? https://github.com/kylecorry31/Trail-Sense/discussions/1940
3. Can you confirm exactly what happens? You select a file, it asks you for the name (and fills it in using the name of the file you selected), then nothing once you click OK?

(the nightly build will install as a separate application that you can use to perform the test and then delete - it won't impact your regular Trail Sense data)

**Reply by SourceOfDeath** [reply]
I am very grateful that you do not give up my problem, it is certainly worth a lot of effort and time for you.
1. Download from F-Droid, link to F-Droid I get in Readme.md from repository on github.
2. Done. Behaves the same way.
3. I launch Trail Sense, Tools (in lowest row) -> Photos of maps -> + -> From file -> standard file selector opens, I navigate to Your pictures -> tap one of them once -> file selector disappear, Immediately appear empty list "photos of maps". That is all.


https://github.com/user-attachments/assets/f147731a-f8c9-4c83-999c-b5f69d523e06

**Reply by kylecorry31** [reply]
Thank you, that is very helpful. I'll keep looking into it.

**Comment by naoliv** [comment]
Hi, @kylecorry31

When I use the signal locator in another city, a "Loading" message stays on the screen longer than usual.
Only two 4G signals appear and no towers are shown.

I know there are cells in OpenCelliD (I’ve captured and uploaded several from that region).
Could the prolonged "Loading" mean the app is trying to fetch data but fails, resulting in no towers being displayed?
Or are you only considering cells captured after a certain date?

I didn’t see any errors in the logcat.
Is there a way to debug this?

Thank you!

**Reply by kylecorry31** [reply]
Hi, the signal finder tool uses a reduced resolution export of the opencellid dataset. That dataset only contains the last 18 months of data (https://opencellid.org/downloads.php), so it is possible if those towers were not reported during that time then they won't be picked up. Also, there is no data fetch from Trail Sense - that data is shipped with the app and was last updated several months ago, so if you didn't report the towers during that time they may not be there. 

Trail Sense only considers towers within 20km from your location as well. It is pretty much impossible for me to debug without an approximate location - if you are comfortable doing so, you can send details to trailsense@protonmail.com (for your privacy though, only give an approximate region and I'll look through the full region for testing, something like 50x50km). If you are not comfortable with that, you can look at the raw data packaged into Trail Sense to see if that lines up. Anything that isn't a fully black pixel is a cell tower: https://github.com/kylecorry31/Trail-Sense/blob/main/app/src/main/assets/cell_towers.webp

(may be hard to see that image, you'll probably want to edit it to make it very bright).

If you are a developer, you can also run this test locally with the coordinates you are interested in to see if it's returning the expected data: https://github.com/kylecorry31/Trail-Sense/blob/main/app/src/androidTest/java/com/kylecorry/trail_sense/tools/signal_finder/CellTowerModelTest.kt (the second parameter to getTowers can be replaced by a Geofence class)

**Reply by naoliv** [reply]
By "fetch" I meant loading from an internal database.
It feels like it tries to load/retrieve the data and, after some time, fails (or times out).
But in fact, according to the [map](https://github.com/kylecorry31/Trail-Sense/blob/main/app/src/main/assets/cell_towers.webp), there are no cells in the area where I am.
Either I captured this data more than 18 months ago, or what I’ve been capturing more recently didn’t exist the last time this data was pulled.
I believe it should be correct in the next update.
Thanks!

**Reply by naoliv** [reply]
By the way, would increasing this time limit (to more than 18 months, for example) make the dataset significantly larger?
I ask because, here in Brazil, I don’t think people capture that much data (and they don’t capture it consistently enough to always have up-to-date information).
Sometimes having older data, if possible, is better than having no data at all.
Thanks again!

**Reply by kylecorry31** [reply]
I'll plan to update the dataset in the October release. It would likely increase the dataset size significantly, but unfortunately I don't have control over that - that 18 month limit is from opencellid, not Trail Sense. See the FAQ of https://opencellid.org/downloads.php (not much details, but provides a little on their rationale). It mentions the API doesn't have that problem, but that requires some additional rules which are not feasible for Trail Sense (either payment or active contributions by Trail Sense) + API is per cell area, not full download like their CSV export.

I'll plan to keep a personal backup of the files and extend it past the 18 month limit (so each updates will be cumulative), but will likely limit that to a few years. #3138
